docs/src/pages/components/pagination/PaginationLink2.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import { MemoryRouter as Router } from 'react-router';
+import { Link as RouterLink } from 'react-router-dom';
+import { makeStyles } from '@material-ui/core/styles';
+import { usePagination } from '@material-ui/lab/Pagination';
+import Pagination from '@material-ui/lab/Pagination';
+import PaginationItem from '@material-ui/lab/PaginationItem';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    '& > *': {
+      marginTop: theme.spacing(2),
+    },
+  },
+}));
+
+// The use of React.forwardRef will no longer be required for react-router-dom v6.
+// See https://github.com/ReactTraining/react-router/issues/6056
+const Link = React.forwardRef((props, ref) => <RouterLink innerRef={ref} {...props} />);
+
+export default function PaginationLink2() {
+  const { items } = usePagination({
```

Comment:
Oh, interesting, for this demo, shouldn't we have a custom Pagination and PaginationItem component not coming from Material-UI?

Reply:
It's primarily focused around router integration. Customisation should probably be separate, and we need to agree the built in styling before adding customisation. I'm not sure customisation requires custom components in any case.